src/App.tsx

```diff
@@ -1,61 +1,102 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { TodoForm } from './components/TodoForm/TodoForm';
 import './App.scss';
+import { getMaxId } from './services/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types';
+import todosFromServer from './api/todos';
+import { getUserBy } from './utils/getUserBy';
+const preparedTodos = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: getUserBy(todo),
+  };
+});
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+type Props = {
+  todos: Todo[];
+};
+export const App: React.FC<Props> = () => {
+  // #region query
+  const [query, setQuery] = useState('');
 
-export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
 
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
+  // #endregion
+  const [todos, setTodos] = useState(preparedTodos);
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => todo.title.includes(query));
+  }, [query, todos]);
+
+  const addTodo = useCallback((todo: Todo) => {
+    setTodos(currentTodos => {
+      const newTodo = {
+        ...todo,
+        id: getMaxId(currentTodos) + 1,
+      };
+
+      return [...currentTodos, newTodo];
+    });
+  }, []);
```

Comment:
The `addTodo` function should include `todos` in its dependency array to avoid potential stale closures. Consider adding `todos` to the dependency array.